Bio/PDB/PDBList.py

```diff
@@ -266,30 +317,78 @@ def update_pdb(self):
+
+        @param pdb_codes: a list of 4-symbols structure Ids from PDB
+        @type pdb_codes: list of strings
+
+        @param file_format: file format. Available options:
+            "mmCif" (default, PDBx/mmCif file),
+            "pdb" (format PDB),
+            "xml" (PMDML/XML format),
+            "mmtf" (highly compressed),
+            "bundle" (PDB formatted archive for large structure}
+
+        @param overwrite: if set to True, existing structure files will be overwritten. Default: False
+        @type overwrite: bool
+
+        @param obsolete: has a meaning only for obsolete structures. If True, download the obsolete structure
+            to 'obsolete' folder, otherwise download won't be performed.
+            This option doesn't work for mmtf format as obsoleted structures doesn't stored in mmtf.
+        (default: False)
+        @type obsolete: bool
+
+        @param pdir: put the file in this directory (default: create a PDB-style directory tree)
+        @type pdir: string
+
+        @return: filenames
+        @rtype: string
+        """
+        for pdb_code in pdb_codes:
+            self.retrieve_pdb_file(pdb_code, obsolete=obsolete, pdir=pdir, file_format=file_format, overwrite=overwrite)
+
+    def download_entire_pdb(self, listfile=None, file_format='mmCif'):
         """Retrieve all PDB entries not present in the local PDB copy.
 
-        Writes a list file containing all PDB codes (optional, if listfile is
-        given).
+        @param listfile: filename to which all PDB codes will be written (optional)
+
+        @param file_format: file format. Available options:
+            "mmCif" (default, PDBx/mmCif file),
+            "pdb" (format PDB),
+            "xml" (PMDML/XML format),
+            "mmtf" (highly compressed),
+            "bundle" (PDB formatted archive for large structure}
+
+        NOTE. The default download format has changed from PDB to PDBx/mmCif
         """
         entries = self.get_all_entries()
         for pdb_code in entries:
-            self.retrieve_pdb_file(pdb_code)
+            self.retrieve_pdb_file(pdb_code, file_format=file_format)
         # Write the list
         if listfile:
             with open(listfile, 'w') as outfile:
                 outfile.writelines((x + '\n' for x in entries))
 
-    def download_obsolete_entries(self, listfile=None):
+    def download_obsolete_entries(self, listfile=None, file_format='mmCif'):
         """Retrieve all obsolete PDB entries not present in the local obsolete
         PDB copy.
 
-        Writes a list file containing all PDB codes (optional, if listfile is
-        given).
+        @param listfile: filename to which all PDB codes will be written (optional)
+
+        @param file_format: file format. Available options:
+            "mmCif" (default, PDBx/mmCif file),
+            "pdb" (format PDB),
+            "xml" (PMDML/XML format),
+
+        NOTE. The default download format has changed from PDB to PDBx/mmCif
```

Comment:
I think it would be worth adding a temporary warning (similar to a deprecation warning) that files are being downloaded in mmCIF. Aside from the legacy grep/awk scripts, I use dozens of executables that only accept PDB format.